src/sync.jl

```diff
@@ -0,0 +1,65 @@
+module Sync
+    global mutex = ReentrantLock()
+    global sync_level = repeat([ 0 ], Threads.nthreads())
```

Comment:
Using `global` on the global level is redundant, isn't it? On the other hand, as we won't reassign these values, making them `const` will help type inference.
```suggestion
    const mutex = ReentrantLock()
    const sync_level = repeat([ 0 ], Threads.nthreads())
```

Reply:
I had added `global` mostly for readability, I think, but I agree that `const` might be better.

Reply:
The `const` is the only way to get type stability. Without it, Julia cannot know that `mutex` always refers to a `ReentrantLock`, and that `sync_level` is always a `Vector{int}`.

Reply:
Sorry, I had already committed the change, but apparently forgot to push it. It's in there now.